video/out/gpu/video.c

```diff
@@ -1381,18 +1384,26 @@ next_bind: ;
     return true;
 }
 
+static struct ra_tex **next_hook_tex(struct gl_video *p)
+{
+    if (p->idx_hook_textures == p->num_hook_textures)
+        MP_TARRAY_APPEND(p, p->hook_textures, p->num_hook_textures, NULL);
```

Comment:
So I assume the intention is to keep hook textures cached across frames, which is why you preserve the array items.

Reply:
Yes, exactly. There are two issues:

1. We can't ra_tex_create() / ra_tex_destroy() every single frame, that would be nuts. So we need to preserve the arrya items, similar to what the old code did
2. We need to keep track of the number of allocated textures separately from the number of “in use in this frame” textures, since the cleanup code needs to destroy them all.

So `idx_hook_textures` works like the old `num_hook_fbos` and `num_hook_textures` works like the old static array size.

Reply:
Sounds fine.

Reply:
Incidentally, it would also be cool if we could somehow associate these with the tex_hook itself instead of needing an extra array. Unfortunately, each tex_hook might still actually require *multiple* ra_tex, since each tex_hook can fire multiple times. (And for pass_opt_hook_point we need a ra_tex for the input *and* the output)

But maybe we could move the entire array to tex_hook instead? In theory, that solves a hypothetical fringe scenario in which hooking OUTPUT + using interpolation at the same time as debanding can end up re-allocating the texture every frame. (Maybe I should test to see if this actually happens?)